executor/executor.go

```diff
@@ -419,56 +424,128 @@ func (e *CheckTableExec) Open(ctx context.Context) error {
 	if err := e.baseExecutor.Open(ctx); err != nil {
 		return errors.Trace(err)
 	}
+	for _, src := range e.srcs {
+		if err := src.Open(ctx); err != nil {
+			return errors.Trace(err)
+		}
+	}
 	e.done = false
 	return nil
 }
 
-// Next implements the Executor Next interface.
-func (e *CheckTableExec) Next(ctx context.Context, chk *chunk.Chunk) error {
+// Close implements the Executor Close interface.
+func (e *CheckTableExec) Close() error {
 	if e.done {
 		return nil
 	}
-	defer func() { e.done = true }()
-	for _, t := range e.tables {
-		dbName := t.DBInfo.Name
-		tb, err := e.is.TableByName(dbName, t.Name)
+
+	close(e.exitCh)
+	var err error
+	for _, src := range e.srcs {
+		err1 := src.Close()
+		if err == nil && err1 != nil {
+			err = err1
+		}
+	}
+	return errors.Trace(err)
+}
+
+func (e *CheckTableExec) checkIndexHandle(ctx context.Context, num int, src *IndexLookUpExecutor) error {
+	cols := src.schema.Columns
+	retFieldTypes := make([]*types.FieldType, len(cols))
+	for i := range cols {
+		retFieldTypes[i] = cols[i].RetType
+	}
+	chk := chunk.New(retFieldTypes, e.initCap, e.maxChunkSize)
+	iter := chunk.NewIterator4Chunk(chk)
+
+	var err error
+	handles := make(map[int64]struct{}, 1024)
```

Comment:
Why use 1024 here?

Reply:
Do you have any suggestions?

Reply:
I don't have any suggestions until I know why it is 1024 here.